src/Shell.c

```diff
@@ -260,6 +262,57 @@ static const char* GetCommandId(CommonData *data)
 }
 
 
+/* retrieve the home directory of the effective user
+ * caller must free returned pointer
+ */
+static char *GetHomeDir()
+{
+    /* cache value of home directory */
+    static char *home = NULL;
```

Comment:
Eh I don't like this. This was originally just going to be a global, but now self owns home. I'm leaning toward rewriting this so self->home is sent to SetHomeDir, which calls GetHomeDir if it's empty.

Okay yeah just give me a minute.